fog/types/src/ingest_common.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2018-2021 The MobileCoin Foundation
+
+//! Contains "native" rust structs that correspond to structs generated by  
+//! protobuf.
+use alloc::{string::String, vec::Vec};
+use mc_crypto_keys::CompressedRistrettoPublic;
+use serde::{Deserialize, Serialize};
+
+/// Mirrors the proto definition of IngestSummary. We have to define this
+/// "native" Rust type to be able to add traits like Serialize to this data.
```

Comment:
I think it's a good idea to use the built-in protobuf JSON once v3 becomes stable. The protobuf-json library seems to have few users and hasn't had a new commit since 2019. IMO, I'd rather have this boilerplate for now then pull a stale third-party dep.

Reply:
Fair enough.